### PR TITLE
Try catch chakracore

### PR DESCRIPTION
**Pull request by gabrielschulhof**
Chakracore version of the exception handling API

### CODE REVIEW

File: src/node_api_helpers.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "node_asyncapi.h"
 #include <limits.h>
 #include <string.h>
+#include <assert.h>
```
Comment (jasongin) [inline src/node_api_helpers.h:17]: Why? I don't see where this is used in the added code.

**Comment by jasongin** [comment]
LGTM

File: src/node_api_helpers.h (hunk 1 above)
Comment (gabrielschulhof) [reply src/node_api_helpers.h:17]: It's used below in existing code and, when you include this header, your code fails to build if you don't make this modification.

Comment (jasongin) [reply src/node_api_helpers.h:17]: OK, I see.

**Pull request closed without merging**